test/token/behaviors/CodexTitle.behavior.js

```diff
@@ -129,46 +129,212 @@ export default function shouldBehaveLikeCodexTitle(accounts) {
       })
     })
 
-    describe('addNewImageHash', function () {
-      const newImageHash = web3.sha3('abc123')
+    describe('modifyMetadataHashes', function () {
+
+      const newNameHash = web3.sha3('New name')
+      const newDescriptionHash = web3.sha3('New description')
+      const newImageHashes = [web3.sha3('New image 1'), web3.sha3('New image 2')]
+
+      describe('when the sender is not authorized', function () {
+        it('should revert', async function () {
+          await assertRevert(
+            this.token.modifyMetadataHashes(
+              0,
+              newNameHash,
+              newDescriptionHash,
+              newImageHashes,
+              providerId,
+              providerMetadataId,
+              { from: unauthorized },
+            ),
+          )
+        })
+      })
 
       describe('when called by the owner', function () {
-        beforeEach(async function () {
-          await this.token.addNewImageHash(0, newImageHash)
+
+        it('should update name hash only', async function () {
+
+          const { logs } = await this.token.modifyMetadataHashes(
+            0,
+            newNameHash,
+            hashedMetadata.description,
+            [],
+            providerId,
+            providerMetadataId,
+          )
+
+          const tokenData = await this.token.getTokenById(0)
+
+          tokenData[0].should.be.equal(newNameHash)
+          tokenData[1].should.be.equal(hashedMetadata.description)
+          tokenData[2].should.deep.equal([hashedMetadata.imageBytes])
+
+          // a Modified event is emitted when provider details are specified
+          logs.length.should.be.equal(1)
+
+          logs[0].event.should.be.eq('Modified')
+          logs[0].args._from.should.be.equal(creator)
+          logs[0].args._tokenId.should.be.bignumber.equal(0)
+          logs[0].args._newNameHash.should.be.equal(tokenData[0])
+          logs[0].args._newDescriptionHash.should.be.equal(tokenData[1])
+          logs[0].args._newImageHashes.should.deep.equal(tokenData[2])
+          logs[0].args._providerId.should.be.equal(providerId)
+          logs[0].args._providerMetadataId.should.be.equal(providerMetadataId)
+
         })
 
-        it('should add the new hash to the imageHashes array', async function () {
+        it('should update description hash only', async function () {
```

Comment:
Can you factor the core logic of these tests into a helper function so it doesn't have to be duped each time? Will make it easier to maintain if we make changes here in the future.